src/EditorFeatures/CSharpTest/CodeActions/Preview/PreviewExceptionTests.cs

```diff
@@ -72,7 +72,7 @@ private static async Task GetPreview(TestWorkspace workspace, CodeRefactoringPro
             var suggestedAction = new CodeRefactoringSuggestedAction(
                 workspace.ExportProvider.GetExportedValue<IThreadingContext>(),
                 workspace.ExportProvider.GetExportedValue<SuggestedActionsSourceProvider>(),
-                workspace, textBuffer, provider, codeActions.First(), fixAllFlavors: null);
+                workspace, workspace.CurrentSolution, textBuffer, provider, codeActions.First(), fixAllFlavors: null);
```

Comment:
in order to be able to tell what changes a lightbulb made, we need the *original* solution at the point the lightbulb items were *computed*.  this involved passign along that piece of state all the way along virally, so you'll see a ton that in this PR.

An alternative was that we could have made it so that anyone who creates an ApplyChangeOperation provided not just the 'final solution' to convert to but also the 'original solution' they started from.  However, this was problematic as every fixer would have to be updated (and it wouldn't affect 3rd party fixers).  So i went with viral approach instead so that all light bulb actions can benefit from this.

Reply:
Note: in tests i'm just passing CurrentSolution along (since tests effectively validate the "there are no intermediary changes" case).  WHen i add new tests that actually validate this new behavior, this will likely need to change.  But those tests are likely to be localized to a small portion of hte test base, so this behavior here for the rest of the tests is fine.